custom_components/pfsense/const.py

```diff
@@ -268,6 +268,7 @@
 SERVICE_START_SERVICE = "start_service"
 SERVICE_STOP_SERVICE = "stop_service"
 SERVICE_RESTART_SERVICE = "restart_service"
+SERVICE_FLUSH_STATES = "flush_states"
```

Comment:
Let's do:

`filter_flush_state_table` and
`filter_flush_nat_table` (if you don't mind adding another service in there while we're at it)

Reply:
flush_states is most accurate in for what I was originally trying to achieve, but then again I just copied what is exposed through the pfsense php under Diagnostics/States/Reset States. 

Flush states includes filter and nat, but flushing nat doesn't include filter. If going into more granularity, you might argue that it's best to have the three separate: nat, rules, and states (both). More code to manage. What are your thoughts?

Reply:
I have an outstanding pull request on the pfsense project to remove filter_flush_nat_table() from filter.inc as it is never used ...

Reply:
filter_flush_state_table() -> pfctl -F states
filter_flush_nat_table() -> pfctl -F nat

pfctl man page says that -F states includes filter and nat tables, I didn't verify in the c code, however I can see that -F nat does not flush the filter table. 

fyi I'm recommending removal of the filter_flush_nat_table() function from filter.inc to the pfsense project as it is never used, and I currently don't have an use case for -F nat

If you think it's a good idea to flush the tables seperately, then it might be useful to have a parameter for the -F argument, but it's more code to manage.

Reply:
```
filter_flush_state_table() calls pfctl -F states
filter_flush_nat_table() calls pfctl -F nat
```

pfctl man page says that -F states includes filter and nat tables, I didn't verify in the c code.

fyi I'm recommending removal of the filter_flush_nat_table() function from filter.inc to the pfsense project as it is never used, 

I'm not sure that I have a use case for -F nat, but if you think it's a good idea to flush the tables separately, then it might be useful to have a parameter for the -F argument, but it's more code to manage. 

question in my mind currently is if -F all is better than -F states?

Reply:
```
filter_flush_state_table() calls pfctl -F states
filter_flush_nat_table() calls pfctl -F nat
```

pfctl man page says that -F states includes filter and nat tables, I didn't verify in the c code.

I'm not sure that I have a use case for -F nat, but if you think it's a good idea to flush rules separately, then it might be useful to have a parameter for the -F argument but it's more code to manage. 

also I question if -F all would be better than -F states?

Reply:
after more digging, `pfctl -F states`  has proven to be the only flush modifier that should be applied after enabled/disabled firewall rules.  With `pfctl -F all` and other variants, firewall configuration is flushed, flushed meaning cleared, and firewall configuration must be reloaded.  

This means, if someone were to call the pfsense filter_flush_nat_table() function which calls `pfctl -F nat` , all translation rules would be deleted but not automatically reloaded.

---

### My use case for reference
I need to remotely enable a firewall rule to turn off streaming access to a specific camera subnet. Enabling the firewall rule from the integration works, but existing connections subject to the rule must be flushed or killed from the pf state table before the enabled firewall rule will take effect.

Since I know the camera subnet of my firewall rule ahead of time, I can use `pfctl -k 0.0.0.0/0 -k camera_subnet` to close only connections to the camera subnet, as opposed to the nuclear option of flushing the entire state table, which closes all open connections on the firewall. 

Closing all connections on the firewall has some nasty side effects to innocent network clients. My next step is to add a kill service to avoid using the nuclear option of clearing the entire state table...

Reply:
I'm satisfied with the original service name upon reconsideration